src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -159,35 +156,101 @@ public void visitJavadocToken(DetailNode ast) {
     }
 
     /**
-     * Finds if inheritDoc is placed properly in java doc.
+     * Checks if period is at the end of sentence.
      * @param ast Javadoc root node.
-     * @return true if inheritDoc is valid or false.
      */
-    private static boolean isOnlyInheritDoc(DetailNode ast) {
-        boolean extraTextFound = false;
-        boolean containsInheritDoc = false;
+    private void checkPeriodAtEnd(DetailNode ast) {
+        final Set<Integer> allowedTypes = new HashSet<>();
+        allowedTypes.add(JavadocTokenTypes.TEXT);
+        allowedTypes.add(JavadocTokenTypes.WS);
```

Comment:
please move `allowedTypes` to class field, it is not going to be changed so has to be ImmutableSet.